### PR TITLE
Replace books count JS alert with modal form

### PR DESCRIPTION
**Pull request by PreetiKotiyal28**
Targets #1765 

### Description
Replace books count JS alert with modal form

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: resources/js/app.js [hunk 1]
```diff
@@ -583,17 +584,19 @@ if (document.getElementById("books_listing")) {
...
+					if (new_count && isFinite(new_count)) {
+						this.books[this.currentIndex].number_of_copies = new_count;
+						axios.put(this.updateRoute + "/" + this.books[this.currentIndex].id, {
+							number_of_copies: new_count
+						});
+					}
+			  },
+			updateIndex: function(index){
+         		 this.currentIndex = index;
+        	},
```
Comment (GauravGusain98) [inline resources/js/app.js:598]:
```suggestion
			updateIndex: function(index){
         		 this.currentIndex = index;
        	         },
```

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 2]
```diff
@@ -67,14 +66,30 @@ class="form-control" id="search_input" placeholder="search all books"
...
-                                    <li @click="updateCopiesCount(index)" class="dropdown-item">Copies Available</li>
+                                    <li @click="updateIndex(index)" data-toggle="modal" data-target="#exampleModal" class="dropdown-item">Copies Available</li>
                                     <li @click="deleteBook(index)" class="dropdown-item text-danger">Delete</li>
                                 </ul>
                             </div>
                         </div>
                     @endcan
                 </div>
-
+                <div class="modal fade" id="exampleModal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
```
Comment (GauravGusain98) [inline resources/views/knowledgecafe/library/books/index.blade.php:76]: Correct the `id` and `arial-labelledby` name here as well

**Review by P4NK4J — CHANGES_REQUESTED** [review]
@PreetiKotiyal28 I tested it out, and overall looks good. It seems like the number of copies available is not preserved in the modal. While its should be populated in the modal as well. Also, the UI of modal seems a bit off, can we have the input field and the label (Number of copies available) in different rows
image

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 3]
```diff
@@ -45,13 +45,12 @@ class="form-control" id="search_input" placeholder="search all books"
         data-index-route="{{ route('books.index') }}" 
         data-category-index-route="{{ route('books.category.index') }}">
         <div class="d-flex flex-wrap w-full">
-            <div v-for="(book, index) in books" class="card book_card  mr-1 mb-3 p-2 mr-lg-4">
-
+            <div v-for="(book, index) in books" :id="'copiesOfBooks'+index" class="card book_card  mr-1 mb-3 p-2 mr-lg-4">
```
[the remark this answers is not in the document]
Comment (GauravGusain98) [reply resources/views/knowledgecafe/library/books/index.blade.php:48]: What it is doing?

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check comments

**Review by P4NK4J — APPROVED** [review]
Looks good.

**Comment by P4NK4J** [comment]
@PreetiKotiyal28 It has some merge conflicts with the index.blade.php please fix them

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 4]
```diff
@@ -67,14 +66,31 @@ class="form-control" id="search_input" placeholder="search all books"
...
-                                    <li @click="updateCopiesCount(index)" class="dropdown-item">Copies Available</li>
+                                    <li @click="updateIndex(index)" data-toggle="modal" :data-target="'#updateModalValue'+index" class="dropdown-item">Copies Available</li>
                                     <li @click="deleteBook(index)" class="dropdown-item text-danger">Delete</li>
                                 </ul>
                             </div>
                         </div>
                     @endcan
                 </div>
-
+                <div class="modal fade" :id="'updateModalValue'+index" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
```
Comment (GauravGusain98) [inline resources/views/knowledgecafe/library/books/index.blade.php:76]:
```suggestion
                <div class="modal fade" :id="'copiesOfBooksCountModal'+index" tabindex="-1" role="dialog" aria-labelledby="copiesOfBooksCount" aria-hidden="true">
```

File: resources/views/knowledgecafe/library/books/index.blade.php [hunk 5]
```diff
@@ -67,14 +66,31 @@ class="form-control" id="search_input" placeholder="search all books"
                                 </a>
                                 <ul class="dropdown-menu ">
                                     <li @click="updateCategoryMode(index)" data-toggle="modal" data-target="#update_category_modal" class="dropdown-item">Update Category</li>
-                                    <li @click="updateCopiesCount(index)" class="dropdown-item">Copies Available</li>
+                                    <li @click="updateIndex(index)" data-toggle="modal" :data-target="'#updateModalValue'+index" class="dropdown-item">Copies Available</li>
```
Comment (GauravGusain98) [inline resources/views/knowledgecafe/library/books/index.blade.php:69]:
```suggestion
                                    <li @click="updateIndex(index)" data-toggle="modal" :data-target="'#copiesOfBooksCountModal'+index" class="dropdown-item">Copies Available</li>
```

Comment (GauravGusain98) [reply resources/views/knowledgecafe/library/books/index.blade.php:69]: I think this name will give more clarity to what this function is doing

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Minor suggestion, else looks Good. Will approve when merge conflicts are resolved

**Comment by GauravGusain98** [comment]
The fixes has been made in another PR. https://github.com/ColoredCow/portal/pull/1936

Closing this

**Pull request closed without merging**